Also parse LC_MESSAGES from environment variables, and also put them at
higher priority than locale.getdefaultlocale()
---
 wicd/misc.py |   12 ++++++++----
 1 file changed, 8 insertions(+), 4 deletions(-)

--- wicd-1.5.9.orig/wicd/misc.py
+++ wicd-1.5.9/wicd/misc.py
@@ -316,6 +316,13 @@ def get_gettext():
     # http://www.learningpython.com/2006/12/03/translating-your-pythonpygtk-application/
     local_path = wpath.translations
     langs = []
+    osLanguage = os.environ.get('LANGUAGE', None)
+    if osLanguage:
+        langs += osLanguage.split(":")
+    osLanguage = None
+    osLanguage = os.environ.get('LC_MESSAGES', None)
+    if osLanguage:
+        langs += osLanguage.split(":")
     try:
         lc, encoding = locale.getdefaultlocale()
     except ValueError, e:
@@ -323,10 +330,7 @@ def get_gettext():
         print "Default locale unavailable, falling back to en_US"
         lc = None
     if lc:
-        langs = [lc]
-    osLanguage = os.environ.get('LANGUAGE', None)
-    if osLanguage:
-        langs += osLanguage.split(":")
+        langs += [lc]
     langs += ["en_US"]
     lang = gettext.translation('wicd', local_path, languages=langs, 
                                fallback=True)
